docs/4_building.md

```diff
@@ -0,0 +1,38 @@
+# 🛠️ Building from source
+
+This page will guide you through building ReVanced Manager from source.
+
+1. Setup the Flutter environment for your [platform](https://docs.flutter.dev/get-started/install)
+
+2. Clone the repository
+
+   ```sh
+   git clone https://github.com/revanced/revanced-manager.git && cd revanced-manager
+   ```
+
+3. Create a GitHub PAT with the `read:packages` scope [here](https://github.com/settings/tokens/new?scopes=read:packages&description=ReVanced) and add your token in `android/gradle.properties`
+
+   ```properties
+   gpr.user = YourUsername
+   gpr.key = ghp_longrandomkey
+   ```
+
+4. Get dependencies
+
+   ```sh
+   flutter pub get
+   ```
+
+5. Delete conflicting outputs
+
+   ```sh
+   flutter packages pub run build_runner build --delete-conflicting-outputs
+   ```
+
+   > **Note**: Must be run every time you sync your local repository with the remote repository
+
+6. Build the APK
+
+   ```sh
+   flutter build apk
+   ```
```

Comment:
3. Create a GitHub personal access token with the `read:packages` scope [here](https://github.com/settings/tokens/new?scopes=read:packages&description=ReVanced)

4. Add your GitHub username and token in `android/gradle.properties`

   ```properties
   gpr.user = YourUsername
   gpr.key = ghp_longrandomkey
   ```

5. Get dependencies

   ```sh
   flutter pub get
   ```

6. Delete conflicting outputs

   ```sh
   flutter packages pub run build_runner build --delete-conflicting-outputs
   ```

   > **Note**: Must be run every time you sync your local repository with the remote repository.

7. Build the APK

   ```sh
   flutter build apk
   ```

Reply:
Something is wrong with the way the suggestion renders so I can't use it idk just edit it and use that

Reply:
What do you mean?

Reply:
```suggestion
3. Create a GitHub personal access token with the `read:packages` scope [here](https://github.com/settings/tokens/new?scopes=read:packages&description=ReVanced)

4. Add your GitHub username and the token to `~/.gradle/gradle.properties`

   ```properties
   gpr.user = YourUsername
   gpr.key = ghp_longrandomkey
   ```

5. Get dependencies

   ```sh
   flutter pub get
   ```

6. Delete conflicting outputs

   ```sh
   flutter packages pub run build_runner build --delete-conflicting-outputs
   ```

   > **Note**: Must be run every time you sync your local repository with the remote repository.

7. Build the APK

   ```sh
   flutter build apk
   ```
```

Reply:
<img width="1407" alt="image" src="https://github.com/revanced/revanced-manager/assets/47723802/32b4700d-fdbf-47eb-b81f-7299db39fc3c">
Yeah GitHub deleted all of this

Reply:
Weird, but looks good now.